mps-model-adapters/src/main/kotlin/org/modelix/model/mpsadapters/MPSRepositoryAsNode.kt

```diff
@@ -62,6 +66,47 @@ data class MPSRepositoryAsNode(val repository: SRepository) : IDefaultNodeAdapte
             emptyList()
         }
     }
+
+    override fun addNewChild(role: IChildLink, index: Int, concept: IConcept?): INode {
+        throw IllegalArgumentException("Template node required for creating MPS modules")
+    }
+
+    override fun addNewChild(role: IChildLink, index: Int, concept: IConceptReference?): INode {
+        throw IllegalArgumentException("Template node required for creating MPS modules")
+    }
+
+    override fun addNewChild(role: String?, index: Int, concept: IConcept?): INode {
+        throw IllegalArgumentException("Template node required for creating MPS modules")
+    }
+
+    override fun addNewChild(role: String?, index: Int, concept: IConceptReference?): INode {
+        throw IllegalArgumentException("Template node required for creating MPS modules")
+    }
+
+    override fun addNewChild(role: IChildLink, index: Int, templateNode: INode): INode {
```

Comment:
What's the reason here that module can only be created from templates?

Reply:
Added a comment.